### PR TITLE
Issue 465

### PR DESCRIPTION
**Pull request by koheiw**
The original Blaheta-Johnson method is added to the `sequence()`.

### CODE REVIEW

**Comment by kbenoit** [comment]
You can resolve the conflicts by merging in master and sorting out the conflicts, then the tests will restart.

File: R/sequences.R [hunk 1]
```diff
@@ -10,7 +10,10 @@
 #' @inheritParams valuetype
 #' @param case_insensitive ignore case when matching, if \code{TRUE}
 #' @param count_min minimum frequency of sequences
-#' @param nested collect nested sequences
+#' @param len_max maxium length of sequences
+#' @param nested if true, collect nested sequences
```
Comment (kbenoit) [inline R/sequences.R]: Can you elaborate here? Not clear to me what nested sequences are or what this argument type is (logical I am guessing)

File: R/sequences.R [hunk 2]
```diff
@@ -10,7 +10,10 @@
...
+#' @param ordered if true, use the Blaheta-Johnson method that distinguishs between 
+#'   the order of tokens, and promotes rare sequences. 
 #' @examples
```
Comment (kbenoit) [inline R/sequences.R]: Can we add an example with the `nested = TRUE`? Also some tests that use this new argument and test for the expected behaviour?

File: R/sequences.R [hunk 3]
```diff
@@ -20,7 +23,8 @@
 #' head(seqs, 10)
 #' 
 #' # types can be any words
-#' seqs2 <- sequences(toks, "^([a-z]+)$", valuetype="regex", case_insensitive = FALSE, count_min = 10)
+#' seqs2 <- sequences(toks, "^([a-z]+)$", valuetype="regex", case_insensitive = FALSE,
```
Comment (kbenoit) [inline R/sequences.R]: and tests for `ordered`

File: R/sequences.R [hunk 4]
```diff
@@ -31,15 +35,15 @@
...
+                      case_insensitive = TRUE, count_min = 2, len_max= 5, nested=TRUE, ordered=FALSE) {
     UseMethod("sequences")
 }
 
 #' @rdname sequences
 #' @noRd
 #' @export
 sequences.tokens <- function(x, features, valuetype = c("glob", "regex", "fixed"),
-                      case_insensitive = TRUE, count_min = 2, nested=TRUE) {
+                      case_insensitive = TRUE, count_min = 2, len_max= 5, nested=TRUE, ordered=FALSE) {
```
Comment (kbenoit) [inline R/sequences.R]: for consistency with the min and max arguments in `dfm_select` and `tokens_select` let's change these arguments to `min_count` and `max_length`

**Comment by koheiw** [comment]
This is only for `sequences` not for `ngrams`.

File: R/sequences.R (hunk 3 above)
Comment (koheiw) [inline R/sequences.R]: Done

File: R/sequences.R (hunk 2 above)
Comment (koheiw) [inline R/sequences.R]: Done

File: R/sequences.R (hunk 1 above)
Comment (koheiw) [inline R/sequences.R]: Done

**Comment by kbenoit** [comment]
Go ahead and merge the PR when you are ready, and add some detail in the commit message for the changes (new features and bug fixes) that I can add to `NEWS.md`. Or you can modify`NEWS.md` directly, but that might require another review if you change this PR.

**Comment by koheiw** [comment]
This branch adds new functions to `sequences()` that are available through `ordered` and `max_length` arguments. When `ordered` is true, orders of words in sequences are distinguished as per the Blaheta-Johnson method. Distinguishing between orders of words leads to creation of very long and sparse vectors, that can causes memory leak when the long sequences are found. To prevent this, `max_length`, which is set to five by default, is added. `max_length` also allows users to extract sequences safely without paddings inserted.

**Pull request merged**